### PR TITLE
 [E2E test] Add Cypress test - Users

### LINKED ISSUE #907: 🧪 [E2E test] Add Cypress test - Users
**_Before you start working on an issue please take a look at our [dedicated hacktoberfest site](https://hacktoberfest.sensenet.com/#eventDetails) to learn the basic rules and find additional resources to help you get started._**

# 🧪E2E test cases

The scope of these tests is to ensure that Users tab in Users and groups menu works as it is intended.

image

## 😎 Role

All test should run as admin.

# Test case 1

## 🧫 Purpose of the test

To ensure that users list has the appropriate data.

## 🐾 Steps

1. Login with admin role
2. **Step:** Click on Users and Groups menu item
    **Expected result:** The users table has the following contents with the following columns:

Contents:
- Business Cat
- Developer Dog

Columns: 
- DisplayName
- Email
- AllRoles
- Enabled
- Actions

# Test case 2

## 🧫 Purpose of the test

To ensure that right click on a user opens context-menu

## 🐾 Steps

1. **Step:** Right click on Developer Dog
**Expected result:** The context-menu dropdown should be visible with the following actions:

- Browse
- Copy to
- Edit
- Move to
- Versions
- Share (Disabled)
- Delete
- Set permissions

2. Click away

# Test case 3

## 🧫 Purpose of the test

Double click on user should open a edit form of the content.

## 🐾 Steps

1. **Step:** Double click on Developer Dog
**Expected result:** Edit form of the user has opened with the title of 'Edit Developer Dog'

image

2. Click Cancel button

### PR DESCRIPTION
**Pull request by ashutosh97**
Resolves #907

### CODE REVIEW

**Comment by ashutosh97** [comment]
@herflis Please review.

File: apps/sensenet/cypress/integration/users-and-groups.spec.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import { pathWithQueryParams } from '../../src/services/query-string-builder'
+
+describe('Users menu', () => {
+  beforeEach(() => {
+    cy.login()
+    cy.visit(pathWithQueryParams({ path: '/', newParams: { repoUrl: Cypress.env('repoUrl') } }))
+  })
+
+  it('ensures that users list has the appropriate data', () => {
+    cy.get('[data-test="Users and groups"]').as('UsersAndGroupsIcon')
```
Comment (herflis) [inline apps/sensenet/cypress/integration/users-and-groups.spec.ts:10]:
```suggestion
    cy.get('[data-test="drawer-menu-item-Users and groups"]').as('UsersAndGroupsIcon')
```

File: apps/sensenet/cypress/integration/users-and-groups.spec.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    cy.get('@ColumnHeaders').children('.Actions').should('exist')
+
+    // Checks for users
+    cy.get('.ReactVirtualized__Table .ReactVirtualized__Table__Grid .ReactVirtualized__Table__row').as('UserRows')
+    cy.get('@UserRows').contains('Business Cat').should('have.length', 1)
+    cy.get('@UserRows').contains('Developer Dog').should('have.length', 1)
+  })
+
+  it('ensures that right click on a user opens context-menu', () => {
+    cy.get('[data-test="Users and groups"]').as('UsersAndGroupsIcon')
```
Comment (herflis) [inline apps/sensenet/cypress/integration/users-and-groups.spec.ts:30]:
```suggestion
    cy.get('[data-test="drawer-menu-item-Users and groups"]').as('UsersAndGroupsIcon')
```

File: apps/sensenet/cypress/integration/users-and-groups.spec.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    cy.get('[data-test="content-context-menu-edit"]').as('EditAction').should('exist')
+    cy.get('[data-test="content-context-menu-moveto"]').as('MoveToAction').should('exist')
+    cy.get('[data-test="content-context-menu-delete"]').as('DeleteAction').should('exist')
+
+    // Click away
+    cy.get('body').click(0, 0)
+  })
+
+  it('ensures that double click on a user opens an edit form of the content', () => {
+    cy.get('[data-test="Users and groups"]').as('UsersAndGroupsIcon')
```
Comment (herflis) [inline apps/sensenet/cypress/integration/users-and-groups.spec.ts:51]:
```suggestion
    cy.get('[data-test="drawer-menu-item-Users and groups"]').as('UsersAndGroupsIcon')
```

File: apps/sensenet/cypress/integration/users-and-groups.spec.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+    // Select Developer Dog
+    cy.get('@UserRows').contains('Developer Dog').dblclick()
+
+    // Check if the required title exist
+    cy.get('[data-test="viewtitle"]').as('Title').should('contain.text', 'Edit Developer Dog')
+    cy.get('form').within(($form) => {
+      cy.get('input[name="FullName"]').should('have.value', 'Developer Dog')
+    })
+    cy.get('[aria-label="Cancel"]').click()
```
Comment (herflis) [inline apps/sensenet/cypress/integration/users-and-groups.spec.ts:65]:
```suggestion
    cy.get('[data-test="cancel"]').click()
```

**Review by herflis — CHANGES_REQUESTED** [review]

**Comment by herflis** [comment]
@ashutosh97 thank you for taking the time to help us! We greatly appreciate your contribution, you will get the extra sensenet-Hacktoberfest swag 🧦. We will contact you via email after the Hacktoberfest event. Hope you are having a great Hacktoberfest! 🎉 🙂

**Pull request merged**